yt_dlp/extractor/nebula.py

```diff
@@ -0,0 +1,320 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import json
+import os
+
+from .common import InfoExtractor
+from ..compat import compat_urllib_parse_unquote, compat_str
+from ..utils import parse_iso8601, ExtractorError, try_get, urljoin, sanitized_Request
+
+
+class NebulaIE(InfoExtractor):
+    """
+    Nebula (https://watchnebula.com/) is a video platform created by the streamer community Standard. It hosts videos
+    off-YouTube from a small hand-picked group of creators.
+
+    All videos require a subscription to watch. There are no known freely available videos. An authentication token to
+    an account with a valid subscription can be specified in multiple ways, including credentials in .netrc or a cookie
+    jar.
+    As neither of these parameters appear to be supported by the unit test runner, it's recommended to set the envvar
+    NEBULA_TOKEN to execute the test runs.
+
+    Nebula uses the Zype video infrastructure and this extractor is using the 'url_transparent' mode to hand off
+    video extraction to the Zype extractor.
+
+    This description has been last updated on 2020-10-22.
+    """

```

Comment:
remove

Reply:
Are you willing to negotiate that?

I think this class documentation gives a good overview of what the extractor is for. ('Nebula' does not seem a particularly globally unique name, even if I'm not aware of another platform with that name today.)

And then it gives a broad introduction to how the extractor works. I think that's a good stepping stone for anyone else going into this code, and should ease code understanding by a decent amount.

Of course I'll update this as the PR progresses, e.g. the removed alternate authentication methods.